src/driver/driver.rs

```diff
@@ -0,0 +1,45 @@
+//! The driver trait which all drivers will implement.
+
+use schema::Definition;
+use error::{Error, ErrorCode};
+use patch::Patch;
+use value::{Pointer, Value};
+use query::Query;
+
+pub trait Driver {
+  /// Connects to a driver and returns a driver instance. After calling this
+  /// the driver is ready to roll!
+  fn connect(uri: &str) -> Result<Box<Self>, Error>;
```

Comment:
Makes sense. I'm going to change the return type to `Result<Self, Error>` with this in mind. In theory the `Box<Driver>` really needs to be put wherever we dynamically load drivers, not here. Also, I'm getting compiler error `E0038` with `Result<Box<Driver>, Error>`.